### ISSUE TITLE
instagram.js could be more efficient

### ISSUE THREAD

**Issue by saiichihashimoto** [marcoajonker/poptags#4]
Right now, instagram.js a big async.waterfall. The first function uses async.map to call ig.tag_search on each trend to get an instagram tag for each twitter trend. The next function uses another async.map to get those tags and run ig.tag_media_recent on each 50 times so we can change each tag into 100 pictures. In other words, you have one map feeding into another map. Cool.

The problem here is that ig.tag_search could run really quickly for one ig.tag_search but not for some others, but that first result won't move onto its 50 ig.tag_media_recents, because nothing in the second async.map will run until the first one is completely done.

TLDR; Have an async.map that has an async.waterfall inside of it instead of the other way around so each async.map isn't blocking the one before it.

PS. If one ig.tag_search or ig.tag_media_recent errors, you want to throw away that result, but not all of them, right? Right now, an error anywhere will tell everything to quit.

**Comment by saiichihashimoto** [comment]
No one cares and I don't want my issues cluttered so bye bye.
